syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package TensorTransmit;

// The tensor transmission service definition.
service TensorTransmit {
  // Sends a tensor
  rpc GetActivationFloat (Layer_f) returns (ActivationFloat) {}

  rpc GetActivationByte (Layer_b) returns (ActivationByte) {}

}

// The request message containing the index of the layer.
message Layer_f {
  int32 layer_index = 1;
}

message Layer_b {
  int32 layer_index = 1;
  string desired_dtype = 2;
}

// The response message containing the tensor and the shape
message ActivationFloat {
  repeated float tensor = 1;
  string shape_f = 2;
}

message ActivationByte {
  bytes buffer = 1;
  string shape_b = 2;
  string dtype = 3;
}